modules/packages.xqm

```diff
@@ -282,10 +283,12 @@ declare function packages:get-package-meta($app as xs:string, $name as xs:string
 (: should be private but there seems to be a bug :)
 declare function packages:public-repo-contents($installed as element(repo-app)*) {
     try {
-        let $url := $config:DEFAULT-REPO || "/public/apps.xml?version=" || packages:get-version() ||
+        for $pkgs in $packages:repos
+        let $urls := $pkgs || "/public/apps.xml?version=" || packages:get-version() ||
             "&amp;source=" || util:system-property("product-source")
         (: EXPath client module does not work properly. No idea why. :)
-        let $request :=
+        let $request := for $url in $urls
```

Comment:
This `for` doesn't make any sense to me. As a `$pkg` (or `$pkgs`) can only be an item() not a sequence. Therefore I think this `for` is redundant...

Reply:
It is not redundant, without it the sequence of all packages is looked up in both repos, with 1 cannot find package error message for each package from repo a not present in repo b, and vice versa. 

The for loop sends the request to the endpoint it should go to, the repo that constrains the packages in question.